File: server/plugin/ms_clone/router/msDiscussion.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var MsDiscussion = new(msDiscussion)

type msDiscussion struct {}

// Init 初始化 社区讨论 路由信息
func (r *msDiscussion) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
	    group := private.Group("discussion").Use(middleware.OperationRecord())
		group.POST("createMsDiscussion", apiMsDiscussion.CreateMsDiscussion)   // 新建社区讨论
		group.DELETE("deleteMsDiscussion", apiMsDiscussion.DeleteMsDiscussion) // 删除社区讨论
		group.DELETE("deleteMsDiscussionByIds", apiMsDiscussion.DeleteMsDiscussionByIds) // 批量删除社区讨论
		group.PUT("updateMsDiscussion", apiMsDiscussion.UpdateMsDiscussion)    // 更新社区讨论
		group.POST("likeMsDiscussion", apiMsDiscussion.LikeMsDiscussion)       // 点赞社区讨论
	}
	{
		group := private.Group("discussion")
		group.GET("findMsDiscussion", apiMsDiscussion.FindMsDiscussion)        // 根据ID获取社区讨论
		group.GET("getMsDiscussionList", apiMsDiscussion.GetMsDiscussionList)  // 获取社区讨论列表
	}
	{
	    group := public.Group("discussion")
	    group.GET("getMsDiscussionDataSource", apiMsDiscussion.GetMsDiscussionDataSource)  // 获取社区讨论数据源
		group.GET("getMsDiscussionPublic", apiMsDiscussion.GetMsDiscussionPublic)  // 社区讨论开放接口
		group.POST("viewMsDiscussion", apiMsDiscussion.ViewMsDiscussion)       // 增加浏览量(公开接口)
	}
}

File: web/src/plugin/ms_clone/api/ms_discussion.js
import service from '@/utils/request'
// @Tags MsDiscussion
// @Summary 创建社区讨论
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.MsDiscussion true "创建社区讨论"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /discussion/createMsDiscussion [post]
export const createMsDiscussion = (data) => {
  return service({
    url: '/discussion/createMsDiscussion',
    method: 'post',
    data
  })
}

// @Tags MsDiscussion
// @Summary 删除社区讨论
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.MsDiscussion true "删除社区讨论"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /discussion/deleteMsDiscussion [delete]
export const deleteMsDiscussion = (params) => {
  return service({
    url: '/discussion/deleteMsDiscussion',
    method: 'delete',
    params
  })
}

// @Tags MsDiscussion
// @Summary 批量删除社区讨论
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除社区讨论"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /discussion/deleteMsDiscussion [delete]
export const deleteMsDiscussionByIds = (params) => {
  return service({
    url: '/discussion/deleteMsDiscussionByIds',
    method: 'delete',
    params
  })
}

// @Tags MsDiscussion
// @Summary 更新社区讨论
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.MsDiscussion true "更新社区讨论"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /discussion/updateMsDiscussion [put]
export const updateMsDiscussion = (data) => {
  return service({
    url: '/discussion/updateMsDiscussion',
    method: 'put',
    data
  })
}

// @Tags MsDiscussion
// @Summary 用id查询社区讨论
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query model.MsDiscussion true "用id查询社区讨论"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /discussion/findMsDiscussion [get]
export const findMsDiscussion = (params) => {
  return service({
    url: '/discussion/findMsDiscussion',
    method: 'get',
    params
  })
}

// @Tags MsDiscussion
// @Summary 分页获取社区讨论列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取社区讨论列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /discussion/getMsDiscussionList [get]
export const getMsDiscussionList = (params) => {
  return service({
    url: '/discussion/getMsDiscussionList',
    method: 'get',
    params
  })
}
// @Tags MsDiscussion
// @Summary 获取数据源
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /discussion/findMsDiscussionDataSource [get]
export const getMsDiscussionDataSource = () => {
  return service({
    url: '/discussion/getMsDiscussionDataSource',
    method: 'get',
  })
}
// @Tags MsDiscussion
// @Summary 不需要鉴权的社区讨论接口
// @Accept application/json
// @Produce application/json
// @Param data query request.MsDiscussionSearch true "分页获取社区讨论列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /discussion/getMsDiscussionPublic [get]
export const getMsDiscussionPublic = () => {
  return service({
    url: '/discussion/getMsDiscussionPublic',
    method: 'get',
  })
}
